File: Codeforces/Brute Force Practice/PFast_Inc.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

template<typename T>
void get_all_comb(vector<T>& vec, int r, vector<vector<T>>& h) {
    int n = vec.size();

    vector<bool> v(n);
    fill(v.end() - r, v.end(), true);

    do {
        vector<T> to_push;
        for (int i = 0; i < n; ++i) {
            if (v[i]) {
                to_push.push_back(vec[i]);
            }
        }
        h.push_back(to_push);
    } while (next_permutation(v.begin(), v.end()));
}

int main() {
    int n, m;
    int bad_count;
    int num;
    cin >> n >> m;

    vector<string> names;
    vector<pair<string, string>> incomp;
    vector<string> solns;

    string input;
    for (int i = 0; i < n; ++i) {
        cin >> input;
        names.push_back(input);
    }

    string input2;
    for (int i = 0; i < m; ++i) {
        cin >> input >> input2;
        incomp.push_back(make_pair(input, input2));
    }

    if (m == 0) {
        cout << n;
        cout << endl;
        sort(names.begin(), names.end());
        for (vector<string>::iterator it = names.begin(); it != names.end(); ++it) {
            cout << *it << endl;
        }
        return 0;

    } else {

        bool done = false;
        for (int i = n; i > 0; --i) {
            vector<vector<string>> poss;
            get_all_comb(names, i, poss);
            for (vector<vector<string>>::iterator it = poss.begin(); it != poss.end(); ++it) {
                if (!done) {
                    bad_count = 0;
                    for (pair<string, string> p : incomp) {
                        if (!(find(it->begin(), it->end(), p.first) != it->end() && find(it->begin(), it->end(), p.second) != it->end())) {
                            num = it->size();
                        } else {
                            bad_count++;
                        }
                    }
                    if (bad_count == 0) {
                        for (string n : *it) {
                            solns.push_back(n);
                        }
                        done = true;
                    }
                }
            }
        }
        cout << num << endl;
        sort(solns.begin(), solns.end());
        for (string n : solns) { cout << n << endl; }
    }

    return 0;
}

File: Codeforces/Brute Force Practice/PFast_inc.py
import itertools

n, m = map(int, input().split())
names = []
incomp = []
solns = []

for i in range(n):
    names.append(input())

for i in range(m):
    incomp.append(input().split())

if m is 0:
    print(n)
    names.sort()
    for i in names:
        print(i)

else:
    done = False
    for x in range(n, 0, -1):
        poss = list(itertools.combinations(names, x))
        for p in poss:
            if not done:
                bad = 0
                for i in incomp:
                    if (not(i[0] in p and i[1] in p)):
                        num = len(p)
                    else:
                        bad += 1
                if bad is 0:
                    for name in p:
                        solns.append(name)
                        done = True

    print(num)
    solns.sort()
    for name in solns:
        print(name)
